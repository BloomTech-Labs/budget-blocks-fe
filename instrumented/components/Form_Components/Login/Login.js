var cov_1yb9kx6jro = function () {
  var path = "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Form_Components/Login/Login.js";
  var hash = "dd554c105b3f4f5ae7c7ff64c8b3c071bbdd5df9";
  var global = new Function("return this")();
  var gcv = "__coverage__";
  var coverageData = {
    path: "/Users/Cha/Desktop/Lambda/budget-blocks-fe/src/components/Form_Components/Login/Login.js",
    statementMap: {
      "0": {
        start: {
          line: 19,
          column: 21
        },
        end: {
          line: 133,
          column: 1
        }
      },
      "1": {
        start: {
          line: 25,
          column: 26
        },
        end: {
          line: 25,
          column: 63
        }
      },
      "2": {
        start: {
          line: 27,
          column: 23
        },
        end: {
          line: 32,
          column: 3
        }
      },
      "3": {
        start: {
          line: 29,
          column: 4
        },
        end: {
          line: 29,
          column: 65
        }
      },
      "4": {
        start: {
          line: 31,
          column: 4
        },
        end: {
          line: 31,
          column: 43
        }
      },
      "5": {
        start: {
          line: 34,
          column: 23
        },
        end: {
          line: 49,
          column: 3
        }
      },
      "6": {
        start: {
          line: 35,
          column: 4
        },
        end: {
          line: 35,
          column: 23
        }
      },
      "7": {
        start: {
          line: 39,
          column: 18
        },
        end: {
          line: 39,
          column: 48
        }
      },
      "8": {
        start: {
          line: 41,
          column: 4
        },
        end: {
          line: 48,
          column: 5
        }
      },
      "9": {
        start: {
          line: 43,
          column: 6
        },
        end: {
          line: 43,
          column: 30
        }
      },
      "10": {
        start: {
          line: 46,
          column: 6
        },
        end: {
          line: 46,
          column: 43
        }
      },
      "11": {
        start: {
          line: 47,
          column: 6
        },
        end: {
          line: 47,
          column: 43
        }
      },
      "12": {
        start: {
          line: 51,
          column: 30
        },
        end: {
          line: 64,
          column: 4
        }
      },
      "13": {
        start: {
          line: 66,
          column: 2
        },
        end: {
          line: 75,
          column: 13
        }
      },
      "14": {
        start: {
          line: 68,
          column: 4
        },
        end: {
          line: 74,
          column: 5
        }
      },
      "15": {
        start: {
          line: 70,
          column: 6
        },
        end: {
          line: 70,
          column: 60
        }
      },
      "16": {
        start: {
          line: 73,
          column: 6
        },
        end: {
          line: 73,
          column: 59
        }
      },
      "17": {
        start: {
          line: 76,
          column: 2
        },
        end: {
          line: 132,
          column: 4
        }
      },
      "18": {
        start: {
          line: 136,
          column: 2
        },
        end: {
          line: 139,
          column: 4
        }
      }
    },
    fnMap: {
      "0": {
        name: "(anonymous_0)",
        decl: {
          start: {
            line: 19,
            column: 21
          },
          end: {
            line: 19,
            column: 22
          }
        },
        loc: {
          start: {
            line: 19,
            column: 30
          },
          end: {
            line: 133,
            column: 1
          }
        },
        line: 19
      },
      "1": {
        name: "(anonymous_1)",
        decl: {
          start: {
            line: 27,
            column: 23
          },
          end: {
            line: 27,
            column: 24
          }
        },
        loc: {
          start: {
            line: 27,
            column: 28
          },
          end: {
            line: 32,
            column: 3
          }
        },
        line: 27
      },
      "2": {
        name: "(anonymous_2)",
        decl: {
          start: {
            line: 34,
            column: 23
          },
          end: {
            line: 34,
            column: 24
          }
        },
        loc: {
          start: {
            line: 34,
            column: 28
          },
          end: {
            line: 49,
            column: 3
          }
        },
        line: 34
      },
      "3": {
        name: "(anonymous_3)",
        decl: {
          start: {
            line: 66,
            column: 12
          },
          end: {
            line: 66,
            column: 13
          }
        },
        loc: {
          start: {
            line: 66,
            column: 18
          },
          end: {
            line: 75,
            column: 3
          }
        },
        line: 66
      },
      "4": {
        name: "mapStateToProps",
        decl: {
          start: {
            line: 135,
            column: 9
          },
          end: {
            line: 135,
            column: 24
          }
        },
        loc: {
          start: {
            line: 135,
            column: 32
          },
          end: {
            line: 140,
            column: 1
          }
        },
        line: 135
      }
    },
    branchMap: {
      "0": {
        loc: {
          start: {
            line: 41,
            column: 4
          },
          end: {
            line: 48,
            column: 5
          }
        },
        type: "if",
        locations: [{
          start: {
            line: 41,
            column: 4
          },
          end: {
            line: 48,
            column: 5
          }
        }, {
          start: {
            line: 41,
            column: 4
          },
          end: {
            line: 48,
            column: 5
          }
        }],
        line: 41
      },
      "1": {
        loc: {
          start: {
            line: 68,
            column: 4
          },
          end: {
            line: 74,
            column: 5
          }
        },
        type: "if",
        locations: [{
          start: {
            line: 68,
            column: 4
          },
          end: {
            line: 74,
            column: 5
          }
        }, {
          start: {
            line: 68,
            column: 4
          },
          end: {
            line: 74,
            column: 5
          }
        }],
        line: 68
      },
      "2": {
        loc: {
          start: {
            line: 68,
            column: 8
          },
          end: {
            line: 68,
            column: 71
          }
        },
        type: "binary-expr",
        locations: [{
          start: {
            line: 68,
            column: 8
          },
          end: {
            line: 68,
            column: 39
          }
        }, {
          start: {
            line: 68,
            column: 43
          },
          end: {
            line: 68,
            column: 71
          }
        }],
        line: 68
      },
      "3": {
        loc: {
          start: {
            line: 108,
            column: 16
          },
          end: {
            line: 108,
            column: 134
          }
        },
        type: "cond-expr",
        locations: [{
          start: {
            line: 108,
            column: 28
          },
          end: {
            line: 108,
            column: 101
          }
        }, {
          start: {
            line: 108,
            column: 102
          },
          end: {
            line: 108,
            column: 134
          }
        }],
        line: 108
      },
      "4": {
        loc: {
          start: {
            line: 117,
            column: 13
          },
          end: {
            line: 127,
            column: 13
          }
        },
        type: "cond-expr",
        locations: [{
          start: {
            line: 118,
            column: 14
          },
          end: {
            line: 124,
            column: 16
          }
        }, {
          start: {
            line: 126,
            column: 14
          },
          end: {
            line: 126,
            column: 28
          }
        }],
        line: 117
      }
    },
    s: {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 0,
      "5": 0,
      "6": 0,
      "7": 0,
      "8": 0,
      "9": 0,
      "10": 0,
      "11": 0,
      "12": 0,
      "13": 0,
      "14": 0,
      "15": 0,
      "16": 0,
      "17": 0,
      "18": 0
    },
    f: {
      "0": 0,
      "1": 0,
      "2": 0,
      "3": 0,
      "4": 0
    },
    b: {
      "0": [0, 0],
      "1": [0, 0],
      "2": [0, 0],
      "3": [0, 0],
      "4": [0, 0]
    },
    _coverageSchema: "43e27e138ebf9cfc5966b082cf9a028302ed4184",
    hash: "dd554c105b3f4f5ae7c7ff64c8b3c071bbdd5df9"
  };
  var coverage = global[gcv] || (global[gcv] = {});

  if (coverage[path] && coverage[path].hash === hash) {
    return coverage[path];
  }

  return coverage[path] = coverageData;
}();

import React, { useState, useEffect } from "react";
import { connect } from "react-redux";
import { loginUser } from "../../../redux/actions/LoginActions";
import Title from "../Title";
import PasswordField from "../PasswordField";
import Account from "../Account";
import { CheckEmptyFields } from "../CheckEmpyFields";
import { ChangeCheckField } from "../ChangeCheckField";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
import Typography from "@material-ui/core/Typography";
import TextField from "@material-ui/core/TextField";
import FormControl from "@material-ui/core/FormControl";
import Button from "@material-ui/core/Button";
import "./loginStyle.css";
import Container from "@material-ui/core/Container";
import Loader from "react-loader-spinner";
cov_1yb9kx6jro.s[0]++;
export const Login = props => {
  cov_1yb9kx6jro.f[0]++;
  // This Component creates the login form. Two Fields: email and password
  // user is used to save form data for submission.
  // values is an object that has objects for each part of the form
  // each object in values are just details for the field such as making email and password error with a message or the submit button disable 
  const [user, setUser] = (cov_1yb9kx6jro.s[1]++, useState({
    email: "",
    password: ""
  }));
  cov_1yb9kx6jro.s[2]++;

  const handleChange = e => {
    cov_1yb9kx6jro.f[1]++;
    cov_1yb9kx6jro.s[3]++;
    // updates for data with new entry. removes any spaces at the begining and end of the string. 
    setUser({ ...user,
      [e.target.name]: e.target.value.trim()
    }); // updates form field with error and message accordingly

    cov_1yb9kx6jro.s[4]++;
    setValues(ChangeCheckField(e, values));
  };

  cov_1yb9kx6jro.s[5]++;

  const handleSubmit = e => {
    cov_1yb9kx6jro.f[2]++;
    cov_1yb9kx6jro.s[6]++;
    e.preventDefault(); // calls function to see if fields are empty. 
    // if fields are empty: it returns an object with new values.
    // if there are no empty fields: it returns boolean of false

    const check = (cov_1yb9kx6jro.s[7]++, CheckEmptyFields(user, values)); // check for returned object

    cov_1yb9kx6jro.s[8]++;

    if (check instanceof Object) {
      cov_1yb9kx6jro.b[0][0]++;
      cov_1yb9kx6jro.s[9]++;
      // if object returned: setValues with that object
      setValues({ ...check
      });
    } else {
      cov_1yb9kx6jro.b[0][1]++;
      cov_1yb9kx6jro.s[10]++;
      // if no object returned: call loginUser action from redux to login
      props.loginUser(user, props.history);
      cov_1yb9kx6jro.s[11]++;
      setUser({
        email: "",
        password: ""
      });
    }
  };

  const [values, setValues] = (cov_1yb9kx6jro.s[12]++, useState({
    showPassword: false,
    password: {
      error: false,
      helperText: ""
    },
    email: {
      error: false,
      helperText: ""
    },
    button: {
      disabled: false
    }
  }));
  cov_1yb9kx6jro.s[13]++;
  useEffect(() => {
    cov_1yb9kx6jro.f[3]++;
    cov_1yb9kx6jro.s[14]++;

    // Everytime a change is made to the form it will check for any errors in any of the given fields
    if ((cov_1yb9kx6jro.b[2][0]++, values.password.error === false) && (cov_1yb9kx6jro.b[2][1]++, values.email.error === false)) {
      cov_1yb9kx6jro.b[1][0]++;
      cov_1yb9kx6jro.s[15]++;
      // if there are no errors: user can click submit button to login
      setValues({ ...values,
        button: {
          disabled: false
        }
      });
    } else {
      cov_1yb9kx6jro.b[1][1]++;
      cov_1yb9kx6jro.s[16]++;
      // if there is an error: disable the button 
      setValues({ ...values,
        button: {
          disabled: true
        }
      });
    }
  }, [user]);
  cov_1yb9kx6jro.s[17]++;
  return <div className="SignIn">
      <Container maxWidth="sm">
        <Title title="Sign In" />

        <form className="SignInForm" onSubmit={handleSubmit}>
          <FormControl variant="filled">
            <Typography className="label">E-Mail Address</Typography>
            <TextField error={values.email.error} helperText={values.email.helperText} placeholder="E-Mail Address" type="text" name="email" onChange={handleChange} value={user.email} variant="outlined" />
          </FormControl>

          <PasswordField name="password" placeholder="Password" label="Password" error={values.password.error} value={user.password} handleChange={handleChange} helperText={values.password.helperText} />
          <Account message="Need an account?" link="/register" />
          {props.error ? (cov_1yb9kx6jro.b[3][0]++, <p style={{
          display: "inline"
        }} className="errorMessage">{props.error}</p>) : (cov_1yb9kx6jro.b[3][1]++, <p className="errorMessage"></p>)}
          <Button variant="outlined" className="signInBtn" type="submit" disabled={values.button.disabled}>
            {
            /* If the form is submitting data for login, the button will change from 'Sign In' to a loading animation */
          }
            {props.isFetching ? (cov_1yb9kx6jro.b[4][0]++, <Loader type="Puff" color="#00BFFF" height={50} width={50} timeout={10000} //3 secs
          />) : (cov_1yb9kx6jro.b[4][1]++, <p>Sign In</p>)}
          </Button>
        </form>
      </Container>
    </div>;
};

function mapStateToProps(state) {
  cov_1yb9kx6jro.f[4]++;
  cov_1yb9kx6jro.s[18]++;
  return {
    isFetching: state.loginReducer.isFetching,
    error: state.loginReducer.error
  };
}

export default connect(mapStateToProps, {
  loginUser
})(Login);