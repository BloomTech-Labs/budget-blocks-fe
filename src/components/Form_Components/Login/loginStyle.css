@import url("https://fonts.googleapis.com/css?family=Exo&display=swap");

.SignIn {
  display: flex;
  margin: 0 2rem;
  /* padding: 1%;
  border-radius: 1%;
  margin: 2%;
  background-color: white; */
}
.inputText {
  display: flex;
  flex-direction: column;
  margin: 2rem 0 2rem 0;
}
.signInBtn {
  display: flex;
  /* You can set any size you want (in %, px, em, etc) */
}
.sign {
  font-family: Exo !important;
  font-style: normal !important;
  font-weight: 600 !important;
  font-size: 30px !important;
  line-height: 38px !important;
  color: #37474f !important;
}
.label {
  text-align: left;
  font-family: Exo !important;
  font-style: normal !important;
  font-weight: normal !important;
  font-size: 14px !important;
  line-height: 22px !important;
  color: #37474f !important;
}
.account {
  display: flex;
  justify-content: center;
  font-family: Exo !important;
  font-style: normal !important;
  font-weight: normal !important;
  font-size: 14px !important;
  line-height: 22px !important;
  color: #357eeb !important;
  margin: 5%;
}
.account strong {
  color: #357eeb;
}
.password {
  margin-top: 30px;
  text-align: left;
}
.confirmPassword {
  margin-top: 30px;
  text-align: left;
}

.login {
  padding: 5%;
  border-radius: 1%;
  margin: 7%;
}
.SignInForm {
  margin: 2%;
  background-color: white;
  padding: 5%;
  border: #e5e5e5 1px solid;
  border-radius: 5px;
}
