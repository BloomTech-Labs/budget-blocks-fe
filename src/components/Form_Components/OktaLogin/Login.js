// FIXME NOT BEING UTILIZED

// import React, { useEffect } from 'react';
// import * as OktaSignIn from '@okta/okta-signin-widget';
// import '@okta/okta-signin-widget/dist/css/okta-sign-in.min.css';

// import config from './config';

// const Login = () => {
//   useEffect(() => {
//     const { pkce, issuer, clientId, redirectUri, scopes } = config;

//     const widget = new OktaSignIn({
//       baseUrl: issuer.split('/oauth2')[0],
//       clientId,
//       redirectUri,
//       logo: '/react.svg',
//       i18n: {
//         en: {
//           'primaryauth.title': 'Sign in to Budget Blocks!',
//         },
//       },
//       authParams: {
//         pkce,
//         issuer,
//         display: 'page',
//         responseMode: pkce ? 'query' : 'fragment',
//         scopes,
//       },
//     });

//     widget.renderEl(
//       { el: '#sign-in-widget' },
//       () => {},
//       (err) => {
//         throw err;
//       }
//     );
//   }, []);

//   return (
//     <div>
//       <div id='sign-in-widget' />
//     </div>
//   );
// };
// export default Login;
