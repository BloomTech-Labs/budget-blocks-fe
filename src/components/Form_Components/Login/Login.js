import React, { useState, useEffect } from "react";
import { Redirect } from "react-router-dom";

import { connect } from "react-redux";
import { loginUser } from "../../../redux/actions/LoginActions";
import Title from "../Title";
import EmailField from "../EmailField";
import PasswordField from "../PasswordField";
import SpinnerLoading from "../SpinnerLoading";
import Account from "../Account";
import { CheckEmptyFields } from "../CheckEmpyFields";
import { ChangeCheckField } from "../ChangeCheckField";
import "react-loader-spinner/dist/loader/css/react-spinner-loader.css";
import Button from "@material-ui/core/Button";
import Container from "@material-ui/core/Container";
import { PageView } from "../../google_analytics/index.js";

import LANG from "../Lang";
import "./loginStyle.css";

const default_values = {
  showPassword: false,
  password: {
    error: false,
    helperText: "",
  },
  email: {
    error: false,
    helperText: "",
  },
  button: {
    disabled: false,
  },
};
const default_user = { email: "", password: "" };

/**
 * Login (React Component)
 * @param {*} props
 * @renders login page
 */
export const Login = (props) => {
  const [state, setState] = useState({
    values: { ...default_values },
    user: { ...default_user },
  });

  const [initial, setInitial] = useState(sessionStorage.getItem("token"));

  useEffect(() => {
    PageView();
  }, []);
  useEffect(() => {
    if (
      state.values.password.error === false &&
      state.values.email.error === false
    ) {
      setState({
        ...state,
        values: { ...state.values, button: { disabled: false } },
      });
    } else {
      setState({
        ...state,
        values: { ...state.values, button: { disabled: true } },
      });
    }
  }, [state.user]); // refactor this useEffect?

  const handleChange = (e) => {
    setState({
      user: { ...state.user, [e.target.name]: e.target.value.trim() },
      values: ChangeCheckField(e, state.values),
    });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    const check = CheckEmptyFields(state.user, state.values);
    if (check instanceof Object) {
      setState({ ...state, values: { ...check } });
    } else {
      props.loginUser(state.user, props.history);
      setState({ ...state, user: { email: "", password: "" } });
    }
  };

  return (
    <div className="SignIn">
      {initial ? (
        <Redirect to="/onBoard/1" />
      ) : (
        <Container maxWidth="sm">
          <Title title={LANG.SIGN_IN} titleClass="SignInTitle" />

          <form className="SignInForm" onSubmit={handleSubmit}>
            <div className="inputText">
              <EmailField
                values={state.values}
                handleChange={handleChange}
                user={state.user}
                fullWidth={true}
              />
              <PasswordField
                name="password"
                placeholder={LANG.PASSWORD}
                label={LANG.PASSWORD}
                error={state.values.password.error}
                value={state.user.password}
                handleChange={handleChange}
                helperText={state.values.password.helperText}
              />
            </div>
            <Account message={LANG.NEED_AN_ACCOUNT} link="/register" />

            {props.error ? (
              <p style={{ display: "inline" }} className="errorMessage">
                {/* {props.error} */}
              </p>
            ) : (
              <p className="errorMessage"></p>
            )}
            <Button
              variant="outlined"
              className="signInBtn"
              type="submit"
              fullWidth={true}
              small="small"
              disabled={state.values.button.disabled}
            >
              {props.isFetching ? <SpinnerLoading /> : <p>{LANG.SIGN_IN}</p>}
            </Button>
          </form>
        </Container>
      )}
    </div>
  );
};

function mapStateToProps(state) {
  return {
    isFetching: state.loginReducer.isFetching,
    error: state.loginReducer.error,
  };
}

export default connect(mapStateToProps, { loginUser })(Login);
