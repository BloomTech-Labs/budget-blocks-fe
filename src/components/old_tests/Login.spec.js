import React from "react";
import { Login } from "../Form_Components/Login/Login";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import { configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { mount, shallow } from "enzyme";

configure({ adapter: new Adapter() });

test("Login renders correctly", () => {
  let wrapper = shallow(<Login />)
  expect(wrapper.exists()).toBe(true);
});

describe("Changing the text inputs changes the values on screen", () => {
  const callAPI = jest.fn();
  const wrapper = mount(

    <Router>
      <Login loginUser={callAPI} />
    </Router>
  );

  let passInput = wrapper.find("input").first();
  let emailInput = wrapper.find("input").first();

  wrapper
    .find("input")
    .last()
    .simulate("change", { target: { value: "test", name: "password" } });
  wrapper
    .find("input")
    .first()
    .simulate("change", {
      target: { value: "sendhelp@gmail.com", name: "email" },
    });

  passInput = wrapper.find("input").first();
  emailInput = wrapper.find("input").first();

  test("Password Input", () => {
    expect(passInput.props.value).toBe("test");
  });

  test("Email Input", () => {
    expect(emailInput.props.value).toBe("sendhelp@gmail.com");
  });
});

describe("Form calls api when form is filled out", () => {
  const callAPI = jest.fn();


  const wrapper = mount(
    <Router>
      <Login loginUser={callAPI} />
    </Router>
  );

  let passInput = wrapper.find("input").first();
  let emailInput = wrapper.find("input").first();
  let signInButton = wrapper.find("button").first();

  wrapper.find("form").simulate("submit");
  wrapper
    .find("input")
    .last()
    .simulate("change", { target: { value: "test", name: "password" } });
  wrapper
    .find("input")
    .first()
    .simulate("change", {
      target: { value: "sendhelp@gmail.com", name: "email" },
    });

  passInput = wrapper.find("input").first();
  emailInput = wrapper.find("input").first();

  test("Password Input", () => {
    expect(passInput.props.value).toBe("test");
  });
  test("Email Input", () => {
    expect(emailInput.props.value).toBe("sendhelp@gmail.com");
  });
  test("Sign in Button", () => {
    expect(signInButton.props.disabled).toBe(false);
  });

  test("API Called", () => {
    expect(callAPI).toHaveBeenCalled();
  });

});
