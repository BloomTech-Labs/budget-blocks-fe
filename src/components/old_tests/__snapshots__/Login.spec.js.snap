// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="SignIn"
      >
        <div
          class="MuiContainer-root MuiContainer-maxWidthSm"
        >
          <div
            class="logo_name"
          >
            <img
              alt="logo"
              class="logo-reg"
              src="Logo.svg"
            />
            <h2
              class="MuiTypography-root sign MuiTypography-h2"
            >
              Sign in
            </h2>
          </div>
          <form
            class="SignInForm"
          >
            <div
              class="inputText"
            >
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth"
              >
                <p
                  class="MuiTypography-root label MuiTypography-body1"
                >
                  E-mail Address
                </p>
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input"
                      name="email"
                      placeholder="E-mail Address"
                      type="email"
                      value=""
                    />
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-77 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-78"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  
                </div>
              </div>
              <div
                class="MuiFormControl-root"
              >
                <p
                  class="MuiTypography-root label MuiTypography-body1"
                >
                   
                  Password
                   
                </p>
                <div
                  class="MuiFormControl-root MuiTextField-root"
                >
                  <div
                    class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                  >
                    <input
                      aria-invalid="false"
                      class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                      name="password"
                      placeholder="Password"
                      type="password"
                      value=""
                    />
                    <div
                      class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                    >
                      <button
                        aria-label="toggle password visibility"
                        class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                        tabindex="0"
                        type="button"
                      >
                        <span
                          class="MuiIconButton-label"
                        >
                          <svg
                            aria-hidden="true"
                            class="MuiSvgIcon-root"
                            focusable="false"
                            role="presentation"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                            />
                          </svg>
                        </span>
                        <span
                          class="MuiTouchRipple-root"
                        />
                      </button>
                    </div>
                    <fieldset
                      aria-hidden="true"
                      class="PrivateNotchedOutline-root-77 MuiOutlinedInput-notchedOutline"
                      style="padding-left: 8px;"
                    >
                      <legend
                        class="PrivateNotchedOutline-legend-78"
                        style="width: 0.01px;"
                      >
                        <span>
                          ​
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                  
                </div>
              </div>
            </div>
            <div
              class="account"
            >
              <a
                class="links"
                href="/register"
              >
                <p
                  class="MuiTypography-root account MuiTypography-body1"
                >
                  Don't have account?
                </p>
              </a>
            </div>
            <p
              class="errorMessage"
            />
            <button
              class="MuiButtonBase-root MuiButton-root MuiButton-outlined signInBtn MuiButton-fullWidth"
              small="small"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-label"
              >
                <p>
                  Sign in
                </p>
              </span>
              <span
                class="MuiTouchRipple-root"
              />
            </button>
          </form>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="SignIn"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthSm"
      >
        <div
          class="logo_name"
        >
          <img
            alt="logo"
            class="logo-reg"
            src="Logo.svg"
          />
          <h2
            class="MuiTypography-root sign MuiTypography-h2"
          >
            Sign in
          </h2>
        </div>
        <form
          class="SignInForm"
        >
          <div
            class="inputText"
          >
            <div
              class="MuiFormControl-root MuiFormControl-fullWidth"
            >
              <p
                class="MuiTypography-root label MuiTypography-body1"
              >
                E-mail Address
              </p>
              <div
                class="MuiFormControl-root MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input"
                    name="email"
                    placeholder="E-mail Address"
                    type="email"
                    value=""
                  />
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-77 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-78"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
                
              </div>
            </div>
            <div
              class="MuiFormControl-root"
            >
              <p
                class="MuiTypography-root label MuiTypography-body1"
              >
                 
                Password
                 
              </p>
              <div
                class="MuiFormControl-root MuiTextField-root"
              >
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
                >
                  <input
                    aria-invalid="false"
                    class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                    name="password"
                    placeholder="Password"
                    type="password"
                    value=""
                  />
                  <div
                    class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
                  >
                    <button
                      aria-label="toggle password visibility"
                      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd"
                      tabindex="0"
                      type="button"
                    >
                      <span
                        class="MuiIconButton-label"
                      >
                        <svg
                          aria-hidden="true"
                          class="MuiSvgIcon-root"
                          focusable="false"
                          role="presentation"
                          viewBox="0 0 24 24"
                        >
                          <path
                            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78l3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
                          />
                        </svg>
                      </span>
                      <span
                        class="MuiTouchRipple-root"
                      />
                    </button>
                  </div>
                  <fieldset
                    aria-hidden="true"
                    class="PrivateNotchedOutline-root-77 MuiOutlinedInput-notchedOutline"
                    style="padding-left: 8px;"
                  >
                    <legend
                      class="PrivateNotchedOutline-legend-78"
                      style="width: 0.01px;"
                    >
                      <span>
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
                
              </div>
            </div>
          </div>
          <div
            class="account"
          >
            <a
              class="links"
              href="/register"
            >
              <p
                class="MuiTypography-root account MuiTypography-body1"
              >
                Don't have account?
              </p>
            </a>
          </div>
          <p
            class="errorMessage"
          />
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined signInBtn MuiButton-fullWidth"
            small="small"
            tabindex="0"
            type="submit"
          >
            <span
              class="MuiButton-label"
            >
              <p>
                Sign in
              </p>
            </span>
            <span
              class="MuiTouchRipple-root"
            />
          </button>
        </form>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
